docs/pages/router/reference/hooks.mdx

```diff
@@ -0,0 +1,168 @@
+
+## `usePathname`
+
+Returns the currently selected route location without search parameters. e.g. `/acme?foo=bar` -> `/acme`. Segments will be normalized: `/[id]?id=normal` -> `/normal`
+
+<RouteUrl>/profile/baconbrix?extra=info</RouteUrl>
+
+```js app/profile/[user].tsx
+import { Text } from 'react-native';
+/* @info */
+import { usePathname } from 'expo-router';
+/* @end */
+
+export default function Route() {
+  /* @info <b>pathname = "/profile/baconbrix"</b> */
+  const pathname = usePathname();
+  /* @end */
+
+  return <Text>User: {user}</Text>;
+}
+```
+
+## `useLocalSearchParams`
+
+Returns the URL search parameters for the contextually selected route. Refer to the [local vs global search params](/router/reference/search-parameters/#local-vs-global-search-parameters) guide for more info.
+
+<FileTree files={['app/_layout.tsx', 'app/[first]/home.tsx', 'app/[second]/shop.tsx']} />
+
+When `/abc/home` pushes `/123/shop`, `useGlobalSearchParams` would return `{ first: undefined, second: '123' }` on `/app/[first]/home.tsx` because the global URL has changed. However, you may want the params to remain `{ first: 'abc' }` to reflect the context of the screen. In this case, you can use `useLocalSearchParams` to ensure the params `{ first: 'abc' }` are still returned in `/app/[first]/home.tsx`.
+
+<RouteUrl>/profile/baconbrix?extra=info</RouteUrl>
+
+```js app/profile/[user].tsx
+import { Text } from 'react-native';
+/* @info */
+import { useLocalSearchParams } from 'expo-router';
+/* @end */
+
+export default function Route() {
+  /* @info */
+  const { user, extra } = useLocalSearchParams();
+  /* @end */
+  return <Text>User: {user}</Text>;
+}
+```
+
+## `useGlobalSearchParams`
+
+Returns the URL search parameters for the globally selected route. e.g. `/acme?foo=bar` -> `{ foo: "bar" }`.
+
+Refer to the [local vs global search params](/router/reference/search-parameters/#local-vs-global-search-parameters) guide for more info.
+
+<RouteUrl>/profile/baconbrix?extra=info</RouteUrl>
+
+```js app/profile/[user].tsx
+import { Text } from 'react-native';
+/* @info */
+import { useGlobalSearchParams } from 'expo-router';
+/* @end */
+
+export default function Route() {
+  /* @info <b>user=baconbrix</b> & <b>extra=info</b> */
+  const { user, extra } = useGlobalSearchParams();
+  /* @end */
+  return <Text>User: {user}</Text>;
+}
+```
+
+### `Href` type
+
+The `Href` type is a union of the following types:
+
+- **string**: A full path like `/profile/settings` or a relative path like `../settings`.
+- **object**: An object with a `pathname` and optional `params` object. The `pathname` can be a full path like `/profile/settings` or a relative path like `../settings`. The `params` can be an object of key/value pairs.
+
+## `useSegments`
+
+Returns a list of segments for the currently selected route. Segments are not normalized, so they will be the same as the file path. e.g. `/[id]?id=normal` -> `["[id]"]`
```

Comment:
```suggestion
Returns a list of segments for the currently selected route. Segments are not normalized so that they will be the same as the file path. For example, `/[id]?id=normal` -> `["[id]"]`.
```